### ISSUE TITLE
Custom CSS: SVG properties are stripped

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#2133]
In 2188364-t the user reported that in the following CSS, the "fill" property gets stripped:

```
header.mobile-menu .mobile-logo .mobile-toggles .toggle-menu svg {
fill: #fff !Important;
}
```

I think that "fill" is not a valid property that can be used like this, but I'm not entirely certain. The property and the use of CSS for styling SVG elements is something that does indeed exist, but I think it can't be used in a stylesheet used for a website? http://tutorials.jenkov.com/svg/svg-and-css.html

Marking this as a bug for now until someone else can determine whether or not we should be stripping this value (because the value is not meant to be used in this way) or not.

**Comment by jeffgolenski** [comment]
@richardmtl There are a lot of unique CSS properties that SVGs utilize, here are a few:

<table class="dataTable" cellspacing="0">
    <tbody><tr>
        <th>CSS Property</th>
        <th>Description</th>
    </tr>
    <tr>
        <td><code>fill</code></td>
        <td>Sets fill color of the shape.</td>
    </tr>
    <tr>
        <td><code>fill-opacity</code></td>
        <td>Sets fill opacity of the shape.</td>
    </tr>
    <tr>
        <td><code>fill-rule</code></td>
        <td>Sets fill rule of the shape.</td>
    </tr>
    <tr>
        <td><code>marker</code></td>
        <td>Sets marker used along the lines (edges) of this shape.</td>
    </tr>
    <tr>
        <td><code>marker-start</code></td>
        <td>Sets start marker used along the lines (edges) of this shape.</td>
    </tr>
    <tr>
        <td><code>marker-mid</code></td>
        <td>Sets mid marker used along the lines (edges) of this shape.</td>
    </tr>
    <tr>
        <td><code>marker-end</code></td>
        <td>Sets end marker used along the lines (edges) of this shape.</td>
    </tr>
    <tr>
        <td><code>stroke</code></td>
        <td>Sets the stroke (line) color used to draw the outline of this shape.</td>
    </tr>
    <tr>
        <td><code>stroke-dasharray</code></td>
        <td>Sets the stroke (line) dashing used to draw the outline of this shape.</td>
    </tr>
    <tr>
        <td><code><nobr>stroke-dashoffset</nobr></code></td>
        <td>Sets the stroke (line) dash offset used to draw the outline of this shape.</td>
    </tr>
    <tr>
        <td><code>stroke-linecap</code></td>
        <td>
            Sets the stroke (line) line cap used to draw the outline of this shape.
            Valid values are <code>round</code>, <code>butt</code> and <code>square</code>.
        </td>
    </tr>
    <tr>
        <td><code>stroke-miterlimit</code></td>
        <td>Sets the stroke (line) miter limit used to draw the outline of this shape.</td>
    </tr>
    <tr>
        <td><code>stroke-opacity</code></td>
        <td>Sets the stroke (line) opacity used to draw the outline of this shape.</td>
    </tr>
    <tr>
        <td><code>stroke-width</code></td>
        <td>Sets the stroke (line) width used to draw the outline of this shape.</td>
    </tr><tr>
        <td><code>text-rendering</code></td>
        <td>Sets the text-rendering used to draw the outline of this shape.</td>
    </tr></tbody></table>

I'll try and do some testing regarding these properties soon to see what gets stripped out and what doesn't. Certainly something we need to support moving forward since SVGs are becoming very prominent.

**Comment by georgestephanis** [comment]
CSS sanitizing is hard -- and a security concern as well.

They'd need to be added here: https://github.com/Automattic/jetpack/blob/master/modules/custom-css/csstidy/data.inc.php

Alternately -- and perhaps a better solution -- would be to write a short second plugin that adds in svg support, as it will be useless to the vast, vast majority of sites.

**Comment by jeffgolenski** [comment]
@georgestephanis Certainly. I would say that if a use is advanced enough to be adding + modding svgs, they probably know enough to be able to work directly within theme files – but even though it's an edge case, it's going to become less of one, quickly. SVG usage is certainly picking it up.

Creating another plugin is a clever idea, but I also think that if we figured out a clever way to add svg support without it, we'd get some praise for being one of the first css editors anywhere to do so. 

Just a thought. I'll leave this in the developers hands!

**Comment by Ipstenu** [comment]
Is there any update to this?

I've moved from font-icons to SVG (for reasons) and the lack of this being possible means I have to move my CSS out of Jetpack's CSS and into a home-grown solution :(

**Comment by jeherve** [comment]
Fixed in #7171
